### ISSUE TITLE
[feature suggestion] Implement "--setuid <uid>" option

### ISSUE THREAD

**Issue by yurivict** [HelloZeroNet/ZeroNet#472]
I would like to run ZeroNet as some low-privileged user. But when I start zeronet with that user it can't read the tor cookie, because this cookie can only be read by the _tor user and root.

Please implement --setuid <uid> option and make zeronet read the tor cookie before it setuids.

**Comment by HelloZeroNet** [comment]
you can add your user to debian-tor group, so it will able to read it: http://zeronet.readthedocs.io/en/latest/faq/#how-to-make-zeronet-work-with-tor-under-linux

**Comment by yurivict** [comment]
I use FreeBSD. On FreeBSD it isn't set up this way:
```
# ls -l /var/db/tor/control_auth_cookie 
-rw-------  1 _tor  _tor  32 May 29 14:45 /var/db/tor/control_auth_cookie
```

In general it is more flexible to have self-setuid option.

**Comment by HelloZeroNet** [comment]
you can use the CookieAuthFileGroupReadable 0|1 configure option 
via https://www.torproject.org/docs/tor-manual.html.en

**Comment by yurivict** [comment]
Thanks.

I just created the FreeBSD port for ZeroNet, and was asking in this context, not for individual use. Since there is no self-suid I have to change it to _tor:_tor as startup. self-setuid would allow to avoid the need to run as _tor:_tor. Changing the tor configuration isn't feasible for the packaged ZeroNet.

**Comment by unsystemizer** [comment]
I agree with @yurivict. I use a stupid workaround now because of this issue (I start tor as `debian-tor` and then `chown` the file with `<user>:debian-tor` (so the group is still "correct" from the Tor perspective and user is correct from ZeroNet perspective).
My ZeroNet user `<user>` is member of `debian-tor` (group) but that does not help.

**Comment by border0464111** [comment]
control_auth_cookie is a deliberately quite restricted file. Regardless of zeronet, Tor best practice are effectively to add the debian-tor group to tor allowed users AND set CookieAuthFileGroupReadable to 1 . 

That's how they manage who can use Tor on the machine. IMO Setting suid on zeronet side seem like a dirty workaround to me.

Others applications using tor can function without suid. zeronet should be able to do the same.

**Comment by yurivict** [comment]
setupid is also a security feature. It helps to isolate the process.
